packages/preview/tauthesis/0.3.2/template/main.typ

```diff
@@ -0,0 +1,49 @@
+/*** main.typ
+ *
+ * The main document to be compiled. Run
+ *
+ *   typst compile main.typ
+ *
+ * to perform the compilation. If you are writing a multi-file
+ * project, this file is where you need to include your content
+ * files.
+ *
+***/
+
+//// Initialize document type.
+
+#import "tauthesis.typ"
```

Comment:
Templates should import their styles through the package manager (i.e. `#import "@preview/tauthesis:0.3.2"`). As-is, the `tauthesis.typ` file is pasted into the user's project directory when they run `typst init` even though it explicitly warns the user to not edit it. Instead, you should move it to the package's root directory and make it the entrypoint of the package in the `typst.toml`. Imports in other files need to be changed, too.

Reply:
Is there a way to test whether the template works after these changes locally? Is there a specific directory where I could put the `tauthesis` file on my system, where `typst` picks the packages from, or are they somehow baked into the binary?

Reply:
Or do I just push the changes here and the CI takes care of that?

Reply:
Ah, this seems to be covered [here][https://typst.app/blog/2023/package-manager/]:

> When the command line compiler encounters a package import, it searches for the directory `typst/packages/{namespace}/{name}-{version}` within the OS-dependant [data](https://docs.rs/dirs/latest/dirs/fn.data_dir.html) and [cache](https://docs.rs/dirs/latest/dirs/fn.cache_dir.html) directories.

Reply:
And here are the system-specific data directories: <https://docs.rs/dirs/latest/dirs/fn.data_dir.html>.

Reply:
Hmm, making a symbolic link `tauthesis-0.3.2 -> $HOME/tauthesis-repo/` in `$HOME/.local/share/typst/packages/preview/` results in the error
```sh
downloading @preview/tauthesis:0.3.2
error: package not found (searched for @preview/tauthesis:0.3.2)
   ┌─ template/main.typ:15:8
   │
15 │ #import "@preview/tauthesis:0.3.2"
```

Reply:
And this is after moving the `tauthesis.typ` file to project root and making it the entrypoint in `typst.toml`.

Reply:
Ok, this was resolved by having a folder `$HOME/.local/share/typst/packages/preview/tauthesis/0.3.2`. So the hyphen between the package name and version is outdated information.